File: uva_12542.cpp
# include <string>
# include <iostream>
using namespace std;
int main(void){
    bool prime[100001];
    for (int i=2; i<100001; ++i){
        if (!prime[i]){
            for (int j=2*i; j<100001; j+=i)
                prime[j]=1;
        }
    }
    string input;
    while(cin >> input){
        if (input=="0")
            break;
        int ans=0;
        for (int i=0; i<input.length(); ++i){
            int temp=0;
            for (int j=i; j<input.length(); ++j){
                temp*=10;
                temp+=(input[j]-'0');
                if (temp>100000)
                    break;
                else if (!prime[temp])
                    ans=temp>ans?temp:ans;
            }
        }
        cout << ans << endl;
    }
    return 0;
}

File: uva_12542.py
prime=[0 for _ in range(100001)]
for i in range(2, 100001):
    if not prime[i]:
        j=2*i
        while j<100001:
            prime[j]=1
            j+=i
while True:
    data=input()
    if data=='0':
        break
    ans=0
    for i in range(len(data)):
        j=i+1
        while j<=len(data):
            temp=int(data[i:j])
            if temp>100000:
                break
            elif not prime[temp]:
                ans=temp if temp>ans else ans
            j+=1
    print(ans)